ethereum/contracts/relayer/coreRelayer/CoreRelayerDelivery.sol

```diff
@@ -488,17 +502,21 @@ abstract contract CoreRelayerDelivery is CoreRelayerBase, IWormholeRelayerDelive
     }
   }
 
-  function decodeCancelled(
+  function tryDecodeExecuteInstructionError(
     bytes memory revertData
-  ) private pure returns (uint32 gasUsed, uint256 available, uint256 required) {
+  ) private pure returns (uint32 gasUsed, bool knownError) {
     uint offset = 0;
     bytes4 selector;
+
+    if(revertData.length < 4) {
+     return (0, false);
+    }
     (selector, offset) = revertData.asBytes4Unchecked(offset);
-    offset += 28;
-    (gasUsed, offset) = revertData.asUint32Unchecked(offset);
-    (available, offset) = revertData.asUint256Unchecked(offset);
-    (required, offset) = revertData.asUint256Unchecked(offset);
-    assert(offset == revertData.length && selector == Cancelled.selector);
+    if((selector == Cancelled.selector) || (selector == RelayProviderReverted.selector) || (selector == RelayProviderPaymentFailed.selector)) {
+      knownError = true;
+      offset += 28;
+      (gasUsed, offset) = revertData.asUint32Unchecked(offset);
```

Comment:
I think we should have a bounds check for this too. I think I'd just unify the bounds check with the above. We would check that the `revertData` has at least 36 bytes; enough to hold the selector and the ABI encoded `uint32` argument.

Reply:
agreed!